core/src/main/java/org/apache/iceberg/util/Tasks.java

```diff
@@ -273,6 +269,16 @@ public boolean run(Task<I, RuntimeException> task) {
       return !threw;
     }
 
+    private void tryRunOnFailure(I item, Exception failure) {
```

Comment:
We factored this out because we're not allowed to have nested `try` blocks in the same method. This is independent of cyclomatic complexity checks.